apps/ledger-live-desktop/src/newArch/WalletSync/useTrustchainSdk.ts

```diff
@@ -0,0 +1,108 @@
+import { getEnv } from "@ledgerhq/live-env";
+import { getSdk } from "@ledgerhq/trustchain/index";
+import {
+  memberCredentialsSelector,
+  resetTrustchainStore,
+  setTrustchain,
+  trustchainSelector,
+} from "@ledgerhq/trustchain/store";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { useDispatch, useSelector } from "react-redux";
+
+import { withDevice } from "@ledgerhq/live-common/hw/deviceAccess";
+import { from, lastValueFrom } from "rxjs";
+import Transport from "@ledgerhq/hw-transport";
+import { TrustchainMember } from "@ledgerhq/trustchain/types";
+import { Flow, Step } from "~/renderer/reducers/walletSync";
+import { setFlow } from "~/renderer/actions/walletSync";
+
+export function runWithDevice<T>(
+  deviceId: string | undefined,
+  fn: (transport: Transport) => Promise<T>,
+): Promise<T> {
+  return lastValueFrom(withDevice(deviceId || "")(transport => from(fn(transport))));
+}
+
+const defaultContext = { applicationId: 16, name: "Ledger Sync LLD" }; // TODO : get name dynamically depending on the platform
+
+export function useLiveAuthenticate() {
+  const sdk = useTrustchainSdk();
+  const trustchain = useSelector(trustchainSelector);
+  const memberCredentials = useSelector(memberCredentialsSelector);
+
+  if (!trustchain || !memberCredentials) {
```

Comment:
i think we should be able to useLiveAuthenticate with these falsy, it hsouldn't throw from the hook.
there will be error cases where we will need to reset the trustchain (e.g. when member is ejected)

Reply:
as `auth(trustchain: Trustchain, memberCredentials: MemberCredentials): Promise<JWT>;
`
We can't :/